a10_neutron_lbaas/a10_context.py

```diff
@@ -86,7 +88,10 @@ class A10WriteContext(A10Context):
     def __exit__(self, exc_type, exc_value, traceback):
         if exc_type is None and self.device_cfg.get('write_memory', True):
             try:
-                self.client.system.action.write_memory()
+                partition_deleted = getattr(self, "partition_deleted", False)
```

Comment:
I'd suggest setting this to False in __init__, so this weird construct doesn't have to exist.  :-)